Add Custom Babel Transformers to Sentry Metro Plugin page (#12495)
getsentry/sentry-docs: docs/platforms/react-native/manual-setup/metro.mdx

---
title: Metro
description: "Learn about the Metro bundler and how to configure it for your application with Sentry React Native SDK."
sidebar_order: 3
---

> Sentry's React Native SDK package ships with a Sentry Metro Plugin which allows you to automatically generate Debug IDs for your applications' bundles. This is crucial for making source maps work properly with Sentry. The plugin also helps you to annotate React component names so they are available in breadcrumbs or minimalize the bundle size by removing unused SDK features.

Reviewer: ```suggestion
Sentry's React Native SDK package ships with a Sentry Metro Plugin which allows you to automatically generate Debug IDs for your applications' bundles. This is crucial for making source maps work properly with Sentry. The plugin also helps you to annotate React component names so they are available in breadcrumbs and minimize the bundle size by removing unused SDK features.
```

## Configuration

The Sentry React Native SDK allows multiple ways to configure the Sentry Metro Plugin, depending on your current use of the Metro configuration.

> ### Use Sentry Metro Plugin

Reviewer: ```suggestion
### Use the Sentry Metro Plugin
```

const config = getSentryExpoConfig(__dirname);
```

> ### Add Custom Babel Transformer

Reviewer: ```suggestion
### Add a Custom Babel Transformer
```

> If you already have a custom transformer, ensure that Sentry Metro Plugin is applied as the last step. The Sentry Metro Plugin doesn't overwrite the existing configuration but extends or wraps existing properties.

Reviewer: ```suggestion
If you already have a custom transformer, ensure that the Sentry Metro Plugin is applied as the last step. The Sentry Metro Plugin doesn't overwrite the existing configuration, it extends or wraps existing properties.
```
